docs/Radio-hardware-specifications-for-EdgeTX.md

```diff
@@ -98,3 +122,11 @@
 * I2C EEPROM
 * Bluetooth module via UART
 * IMU (e.g. LSM6DS33)
+* I2C port extender
+  * if port extenders are used for inputs, the interrupt pin has to be connected to an interrupt capable pin of the MCU
+
+---
+
+## General Considerations
+* When the radio is equipped with flight mode button (6-Pos), those should be connected in a way that they can be used individually. To save pins for the illumination, LEDs with Neopixel interface can be used
```

Comment:
We might need to reconsider the "individual" digital input requirement. Wiring the inputs via analog input saves I/O and works, the corresponding LEDs should be individually addressable from EdgeTX though.

Reply:
Think this just needs examples of those options... i.e. single analog input using resistor ladder, I2C expander. BTW, Neopixel is Adafruit's name for the WS2812 single line serial protocol ;)

Reply:
I would add something like (please apologise for the approximate english);
- switches whose state can be set by EdgeTX need to provide a visual feedback controllable by EdgeTX (led,...)

Reply:
I like the suggestion, would add that this would be then appropriate only to switches that do have a visual feedback (that it would not be misunderstood that we require all switches to have a visual feedback).

Reply:
```suggestion
* When the radio is equipped with flight mode buttons (6POS / six position switch), they should be connected in a way that they can be used individually (i.e. as customisable switches in EdgeTX). In order to save MCU IO pins, this could be done using a single analog input and resistor ladder, or an I2C expander. These switches need to have visual feedback LEDs that can be controlled by the MCU. To save MCU IO pins, this can could be done using I2C expanders, or WS2812 (aka Neopixel) single serial line LEDs. 
```

Reply:
One more thought - possibly we might not necessarily dictate explicitly here "6"-pos, but handle this way any switch with more than 3 positions (e.g. if a manufacturer wants to do a radio with flight mode switch other than 6 positions).

Reply:
I think there are two rules there: 
- traditional 6POS need to be replaced by customisable switch version
- switches whose state can be set by EdgeTX, including but not limited to customisable switch  version of 6pos, need to provide a visual feedback controllable by EdgeTX (led,...)

Reply:
Btw, I don't see how a single analog (been resistor ladder or else) could be used in that setup

Reply:
Using a resistor ladder, you can have multiple buttons hooked to the one analog, and know which button is pressed via the specific voltage it's press triggers. You then do the physical to logical translation (groups, startup state, etc) you normally do for customisable switches. And light whichever led you need to based on the logical configuration. i.e.If you ignored the led on the emulated 6pos, you could use that as a customisable switch, no? The only problem there is actually that we have no control over the led (well, and no nice interrupt).

Reply:
It also mean you can only push one button at a time, unless you try doggy analog to binary conversions that I think can only lead to issues

Reply:
Not the case e.g. with Paladin EV stick end buttons - the output has 4 distinguishable voltage levels with resistor ladder - no button pressed, button A pressed, button B pressed and buttons A and B pressed simultanously.

Reply:
Thats certainly fine for 2 buttons, more tricky for 6 or more as we have on 6pos (since we are talking flight mode selector scenario)

Reply:
The real question there is how many buttons is it reasonable to expect to be pressed at one time, and what to do if more than that number are pressed. i.e. no different to a button matrix.  If you're saying that we should be able to detect all six or more buttons uniquely at once, then sure... sure there could be problems... but then how many fingers do you have, and who is flying the plane? 🤣

Reply:
Probably 2 simultaneous are fine, but out of how many possible is where adc to binary becomes tricky